### PR TITLE
Ham Store Room plugin

### PR DESCRIPTION
**Pull request by Barragek0**
Show green / red overlays on chests in H.A.M store room, depending on the keys you have
yuqZOKS

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/hamstoreroom/HamPlugin.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	private OverlayManager overlayManager;
+
+	@Inject
+	private HamOverlay hamOverlay;
+
+	@Inject
+	private Client client;
+
+	@Getter(AccessLevel.PACKAGE)
+	GameObject steelObject;
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/hamstoreroom/HamPlugin.java]: These should be private

File: runelite-client/src/main/java/net/runelite/client/plugins/hamstoreroom/HamOverlay.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (client.getLocalPlayer().getWorldLocation().getRegionID() != 10321)
+		{
+			return null;
+		}
+
+		if (hamPlugin.steelObject.getCanvasTilePoly() != null)
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/hamstoreroom/HamOverlay.java]: Access these using the Lombok generated getters (eg. `hamPlugin.getSteelObject()`)

**Comment by ShaunDreclin** [comment]
Your gif shows left clicking to pickpocket the guards, is that included in this?

**Comment by Barragek0** [comment]
> 
> 
> Your gif shows left clicking to pickpocket the guards, is that included in this?

Nope, did ask hydrox about committing that and he referred me to this: https://github.com/runelite/runelite/issues/3693

**Pull request closed without merging**